api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -14,10 +14,18 @@ import (
 )
 
 const (
+	// TODO - for the next API bump; the case on these needs to be changed.
+	// See https://github.com/openshift/hypershift/pull/4538#discussion_r1765165827 for more details.
 	ArchitectureAMD64   = "amd64"
 	ArchitectureS390X   = "s390x"
 	ArchitecturePPC64LE = "ppc64le"
 	ArchitectureARM64   = "arm64"
+	ArchitectureMulti   = "multi"
+
```

Comment:
where's the manfiest prefix coming from?

I'd expect we have something like
```
type payloadArch string

const (
	AMD64   payloadArch = "AMD64"
	PPC64LE payloadArch = "PPC64LE"
	ARM64   payloadArch = "ARM64"
	Multi   payloadArch = "Multi"
)

// toPayloadArch converts a string to payloadArch.
func toPayloadArch(arch string) payloadArch {
	switch strings.ToUpper(arch) {
	case string(AMD64):
		return AMD64
	case string(ARM64):
		return ARM64
	case string(PPC64LE):
		return PPC64LE
	case string("MULTI"):
		return Multi
	default:
		return ""
	}
}
```

Then any comparison must use toArch() or the consts
```
hc.Status.PayloadArch != Multi and hc.Status.PayloadArch != toArch(arch)
```

Reply:
Also why don't we have ArchitectureS390X in the new list?

Reply:
I dropped the previous list